raiden_libs/messages/request_paths.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+from typing import Dict
+import struct
+
+import jsonschema
+from eth_utils import is_address, decode_hex
+
+from raiden_libs.messages.message import Message
+from raiden_libs.properties import address_property
+from raiden_libs.messages.json_schema import REQUEST_PATHS_SCHEMA
+from raiden_libs.utils import eth_verify
+from raiden_libs.types import Address
+
+
+class RequestPaths(Message):
+    """ A message to request a path from PFS. It is sent from a raiden node to the PFS. """
+    def __init__(
+        self,
+        token_network_address: Address,
+        target_address: Address,
+        value: int = 1,
+        num_paths: int = 1,
+        chain_id: int = 1,
+        nonce: int = 0,
+        signature: str = None,
+    ) -> None:
+        super().__init__()
+        assert is_address(token_network_address)
+        assert is_address(target_address)
+
+        self._type = 'RequestPaths'
+        self.token_network_address = token_network_address
+        self.target_address = target_address
+        self.value = value
+        self.num_paths = num_paths
+        self.chain_id = chain_id
+        self.nonce = nonce
+        self.signature = signature
+
+    def serialize_data(self) -> Dict:
+        return {
+            'token_network_address': self.token_network_address,
+            'target_address': self.target_address,
+            'value': self.value,
+            'num_paths': self.num_paths,
+            'chain_id': self.chain_id,
+            'nonce': self.nonce,
+            'signature': self.signature,
+        }
+
+    def serialize_bin(self):
+        """Return RequestPaths serialized to binary"""
+        order = '>20s32s32s8sd'
+        return struct.pack(
+            order,
+            decode_hex(self.token_network_address),
+            self.target_address.to_bytes(32, byteorder='big'),
+            self.value.to_bytes(32, byteorder='big'),
+            self.num_paths.to_bytes(32, byteorder='big'),
+            self.chain_id.to_bytes(32, byteorder='big'),
+            self.nonce.to_bytes(8, byteorder='big')
+        )
+
+    @classmethod
+    def deserialize(cls, data):
+        jsonschema.validate(data, REQUEST_PATHS_SCHEMA)
+        ret = cls(
+            token_network_address=data['token_network_address'],
+            target_address=data['target_address'],
+            value=int(data['value']),
+            num_paths=data['num_paths'],
+            chain_id=data['chain_id'],
```

Comment:
this and chain_id should be ints as well

Reply:
Types are already enforced now.